Backgrounds/AstralDesertSurfaceBGStyle.cs

```diff
@@ -1,37 +1,147 @@
+            if (Main.gameMenu)
+            {
+                num4 -= num3;
+            }
+            pushBGTopHack = num4;
+            pushBGTopHack += num;
+            if (flag2)
+            {
+                pushBGTopHack += num3;
+            }
+            if (flag)
+            {
+                var bgScale = 1.25f;
+                var bgParallax = 0.4;
+                var bgTopY = (int)(backgroundTopMagicNumber * 1800.0 + 1500.0) + (int)scAdj + pushBGTopHack;
+                bgScale *= bgGlobalScaleMultiplier;
+                var bgWidthScaled = (int)((float)CalamityMod.AstralDesertSurfaceMiddle.Width * bgScale);
+                SkyManager.Instance.DrawToDepth(Main.spriteBatch, 1.2f / (float)bgParallax);
+                var bgStartX = (int)(0.0 - Math.IEEERemainder((double)Main.screenPosition.X * bgParallax, bgWidthScaled) - (double)(bgWidthScaled / 2));
+                if (Main.gameMenu)
+                    bgTopY = 320 + pushBGTopHack;
+
+                var bgLoops = Main.screenWidth / bgWidthScaled + 2;
+                if ((double)Main.screenPosition.Y < Main.worldSurface * 16.0 + 16.0)
+                {
+                    for (int i = 0; i < bgLoops; i++)
+                    {
+                        Main.spriteBatch.Draw(CalamityMod.AstralDesertSurfaceMiddle, new Vector2(bgStartX + bgWidthScaled * i, bgTopY + MiddleBGYOffset), new Rectangle(0, 0, CalamityMod.AstralDesertSurfaceMiddle.Width, CalamityMod.AstralDesertSurfaceMiddle.Height), ColorOfSurfaceBackgroundsModified, 0f, default(Vector2), bgScale, SpriteEffects.None, 0f);
+                    }
+                }
+
+                bgScale = 1.31f;
+                bgParallax = 0.43;
+                bgTopY = (int)(backgroundTopMagicNumber * 1950.0 + 1750.0) + (int)scAdj + pushBGTopHack;
+                bgScale *= bgGlobalScaleMultiplier;
+                bgWidthScaled = (int)((float)CalamityMod.AstralDesertSurfaceClose.Width * bgScale);
+                SkyManager.Instance.DrawToDepth(Main.spriteBatch, 1f / (float)bgParallax);
+                bgStartX = (int)(0.0 - Math.IEEERemainder((double)Main.screenPosition.X * bgParallax, bgWidthScaled) - (double)(bgWidthScaled / 2));
+                if (Main.gameMenu)
+                {
+                    bgTopY = 400 + pushBGTopHack;
+                    bgStartX -= 80;
+                }
+
+                bgLoops = Main.screenWidth / bgWidthScaled + 2;
+                if ((double)Main.screenPosition.Y < Main.worldSurface * 16.0 + 16.0)
+                {
+                    for (int i = 0; i < bgLoops; i++)
+                    {
+                        Main.spriteBatch.Draw(CalamityMod.AstralDesertSurfaceClose, new Vector2(bgStartX + bgWidthScaled * i, bgTopY + CloseBGYOffset), new Rectangle(0, 0, CalamityMod.AstralDesertSurfaceClose.Width, CalamityMod.AstralDesertSurfaceClose.Height), ColorOfSurfaceBackgroundsModified, 0f, default(Vector2), bgScale, SpriteEffects.None, 0f);
+                    }
```

Comment:
Can we get actual names for the variables here? Code is unclear at a glance. Comments would help too. I have no idea what this code is intending to do.